### DISCUSSION TITLE
Random seed in agents functions

### DISCUSSION THREAD

**Discussion by daniele-baccega** [FLAMEGPU/FLAMEGPU2#1088] [Q&A] [answered]
Hi,
is it possible to set the random seed inside an agent function? This would be useful to use a multi stream random generator where each agent has its own seed

Thanks!

**Comment by Robadob** [comment, marked as answer]
No.

Random state is per-thread, rather than per-agent within FLAMEGPU, therefore a thread's random state is not attached to a single agent unless a model only has 1 agent type, with 1 state and a fixed number of agents which are never sorted (e.g. not using spatial messaging).

If implementing a model in C/C++ you could implement per-agent random state/generation manually, by having [curand](https://docs.nvidia.com/cuda/curand/index.html) states stored behind a `__device__` global array accessed with an agent's unique ID.
